lib/bandit/pipeline.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Bandit.Pipeline do
+  @moduledoc false
+
+  @type transport_info ::
+          {boolean(), ThousandIsland.Transport.socket_info(),
+           ThousandIsland.Transport.socket_info()}
+  @type request_target :: {scheme(), host(), Plug.Conn.port_number(), path()}
+  @type scheme :: String.t() | nil
+  @type host :: Plug.Conn.host() | nil
+  @type path :: String.t() | :*
+
+  # credo:disable-for-this-file Credo.Check.Design.AliasUsage
+
+  @spec run(
+          Plug.Conn.adapter(),
+          transport_info(),
+          Plug.Conn.method(),
+          request_target(),
+          Plug.Conn.headers(),
+          Bandit.plug()
+        ) ::
+          {:ok, Plug.Conn.adapter()} | {:error, term()}
```

Comment:
It looks like it could also return the websocket info from `maybe_upgrade`. Something more like:
```suggestion
          {:ok, Plug.Conn.adapter()} | {:ok, :websocket, term()} | {:error, term()}
```

Reply:
This isn't the first time that dialyzer has missed a `with` clause like this. Starting to be wary....